core/src/main/java/org/elasticsearch/common/lucene/search/Queries.java

```diff
@@ -179,8 +179,6 @@ public static int calculateMinShouldMatch(int optionalClauseCount, String spec)
             result = calc < 0 ? result + calc : calc;
         }
 
-        return (optionalClauseCount < result ?
-                optionalClauseCount : (result < 0 ? 0 : result));
-
+        return result < 0 ? 0 : result;
```

Comment:
> The boolean query throws an IllegalArgumentException when min should match is a negative value but only on some cases, for instance if you don't have should clauses then the query works.

OK, I didn't know that, it's more complex than I thought. Let's merge this PR as-in then and work on another PR to make minimum_should_match validation more consistent.